### PR TITLE
adapt convolution for complex singals and implement testing

### PR DESCRIPTION
**Pull request by tluebeck**
adapt convolution and implement tests

### CODE REVIEW

File: pyfar/dsp/dsp.py [hunk 1]
```diff
@@ -1656,8 +1656,14 @@ def convolve(signal1, signal2, mode='full', method='overlap_add'):
         res[..., :n_min-1] += res[..., -n_min+1:]
         res = res[..., :n_max]
 
+    if res.dtype.kind == 'c':
+        is_result_complex = True
+    else:
+        is_result_complex = False
```
Comment (f-brinkmann) [inline pyfar/dsp/dsp.py:1662]: Could be shortened
```suggestion
     complex = True if res.dtype.kind == 'c' else False
```

**Review by f-brinkmann — CHANGES_REQUESTED** [review]
Thanks for the implementation. I have two general comments
- Can you add a one line docstrings to the tests, e.g., '''Test dsp.convolve with complex signals'''?
- Should we use signals for testing that actually have an imaginary part?

**Review by f-brinkmann — APPROVED** [review]

**Review by ahms5 — COMMENTED** [review]
Thanks for implementing, looks fine to me, but I just have on question.

File: tests/test_dsp.py [hunk 2]
```diff
@@ -791,6 +801,19 @@ def test_convolve_mode_and_method(method, mode, desired):
     np.testing.assert_allclose(res.time, desired, atol=1e-10)
 
 
+@pytest.mark.parametrize("method", ['overlap_add', 'fft'])
+@pytest.mark.parametrize("mode, desired", [
+    ('full', np.array([[1, -0.5, 0.1, -0.35, -0.05, 0.01]], dtype='complex')),
+    ('cut', np.array([[1, -0.5, 0.1, -0.35]], dtype='complex')),
+    ('cyclic', np.array([[0.95, -0.49, 0.1, -0.35]], dtype='complex'))])
+def test_convolve_mode_and_method_complex(method, mode, desired):
```
Comment (ahms5) [inline tests/test_dsp.py:809]: How about testing complex signals and also one complex and one real signal.

**Review by ahms5 — COMMENTED** [review]
Thanks for the effort. just one comment.

**Review by pingelit — APPROVED** [review]
Thanks for implementing!

**Review by ahms5 — APPROVED** [review]

**Pull request merged**